src/pydicom/filewriter.py

```diff
@@ -377,21 +384,37 @@ def write_numbers(fp: DicomIO, elem: DataElement, struct_format: str) -> None:
 
 def write_OBvalue(fp: DicomIO, elem: DataElement) -> None:
     """Write a data_element with VR of 'other byte' (OB)."""
-    if len(elem.value) % 2:
-        # Pad odd length values
-        fp.write(cast(bytes, elem.value))
-        fp.write(b"\x00")
+
+    if elem.is_buffered:
+        bytes_written = 0
+        buffer = cast(BufferedIOBase, elem.value)
+        with reset_buffer_position(buffer):
+            for chunk in read_buffer(buffer):
```

Comment:
I wonder if the chunk size should be configurable - either globally via a config setting, or via an extra parameter in the write functions.

Reply:
Ah yes, I meant to include that as a Settings option.